### PR TITLE
Adapt the legacy control panel to the new dropdown component

### PR DESCRIPTION
**Pull request by brboi**
https://github.com/odoo/enterprise/pull/20682

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/google_drive/static/tests/gdrive_test.js [hunk 1]
```diff
@@ -62,12 +60,12 @@ odoo.define('google_drive.gdrive_integration', function (require) {
...
                 },
             });
-            await cpHelpers.toggleActionMenu(form);
+            await testUtils.controlPanel.toggleActionMenu(form);
 
             assert.containsOnce(form, '.oe_share_gdoc_item',
                 "The button to the google action should be present");
 
-            await cpHelpers.toggleMenuItem(form, "Cyberdyne Systems");
+            await testUtils.controlPanel.toggleMenuItem(form, "Cyberdyne Systems");
```
Comment (Polymorphe57) [inline addons/google_drive/static/tests/gdrive_test.js:68]: why is testUtils.controlPanel still used here?

File: addons/web/static/src/legacy/xml/pivot.xml [hunk 2]
```diff
@@ -133,10 +133,10 @@
         </div>
     </t>
 
-    <t t-name="web.legacy.PivotCustomGroupByItem" t-inherit="web.Legacy.CustomGroupByItem" t-inherit-mode="primary" owl="1">
-        <xpath expr="//div[hasclass('o_generator_menu')]" position="attributes">
+    <t t-name="web.legacy.PivotCustomGroupByItem" t-inherit="web.CustomGroupByItem" t-inherit-mode="primary" owl="1">
+        <!-- <xpath expr="//div" position="attributes">
```
Comment (Polymorphe57) [inline addons/web/static/src/legacy/xml/pivot.xml:137]: change that

File: addons/web/static/tests/legacy/components/action_menus_tests.js [hunk 3]
```diff
@@ -63,7 +62,7 @@ odoo.define('web.action_menus_tests', function (require) {
             assert.strictEqual(dropdowns[1].querySelector('.o_dropdown_title').innerText.trim(), "Action");
             assert.containsNone(actionMenus, '.o_dropdown_menu');
 
-            await cpHelpers.toggleActionMenu(actionMenus, "Action");
+            await testUtils.controlPanel.toggleActionMenu(actionMenus, "Action");
```
Comment (Polymorphe57) [inline addons/web/static/tests/legacy/components/action_menus_tests.js:65]: same question here?

File: addons/web/static/tests/legacy/components/pager_tests.js [hunk 4]
```diff
@@ -27,12 +26,12 @@ odoo.define('web.pager_tests', function (require) {
                 },
             });
 
-            assert.strictEqual(cpHelpers.getPagerValue(pager), "1-4",
+            assert.strictEqual(testUtils.controlPanel.getPagerValue(pager), "1-4",
```
Comment (Polymorphe57) [inline addons/web/static/tests/legacy/components/pager_tests.js:29]: again

File: addons/web/static/tests/legacy/control_panel/search_bar_tests.js [hunk 5]
```diff
@@ -199,7 +193,7 @@ QUnit.module("Search Bar (legacy)", (hooks) => {
 
             assert.containsNone(searchBar, '.o_searchview_autocomplete');
 
-            await cpHelpers.editSearch(searchBar, "Hello there");
+            await testUtils.controlPanel.editSearch(searchBar, "Hello there");
```
Comment (Polymorphe57) [inline addons/web/static/tests/legacy/control_panel/search_bar_tests.js:196]: same here

File: addons/web/static/tests/legacy/fields/relational_fields/field_one2many_tests.js [hunk 6]
```diff
@@ -1845,14 +1845,14 @@ QUnit.module('fields', {}, function () {
             // move to record 2, which has 3 related records (and shouldn't contain the
             // related records of record 1 anymore). Two additional RPCs should have
             // been done
-            await cpHelpers.pagerNext(form);
+            await testUtils.controlPanel.pagerNext(form);
```
Comment (Polymorphe57) [inline addons/web/static/tests/legacy/fields/relational_fields/field_one2many_tests.js:1848]: again

File: addons/web/static/tests/legacy/widgets/data_export_tests.js [hunk 7]
```diff
@@ -114,8 +113,8 @@ QUnit.module('widgets', {
 
         await testUtils.dom.click(list.$('thead th.o_list_record_selector input'));
 
-        await cpHelpers.toggleActionMenu(list);
-        await cpHelpers.toggleMenuItem(list, 'Export');
+        await testUtils.controlPanel.toggleActionMenu(list);
+        await testUtils.controlPanel.toggleMenuItem(list, 'Export');
```
Comment (Polymorphe57) [inline addons/web/static/tests/legacy/widgets/data_export_tests.js:117]: same here

**Comment by Polymorphe57** [comment]
Convert ActionMenu or not?

File: addons/web/static/tests/legacy/control_panel/custom_filter_item_tests.js [hunk 8]
```diff
@@ -520,8 +515,8 @@ odoo.define('web.filter_menu_generator_tests', function (require) {
                             type: "filter",
                         },
                         {
-                            description: 'Boolean Field is true',
-                            domain: '[["boolean_field","=",True]]',
+                            description: 'Boolean Field is false',
```
[the remark this answers is not in the document]
Comment (brboi) [reply addons/web/static/tests/legacy/control_panel/custom_filter_item_tests.js:518]: the test was not correct at first, see below how the boolean_field condition is set

File: addons/web/static/src/legacy/xml/pivot.xml [hunk 9]
```diff
@@ -75,68 +75,53 @@
...
-                           class="dropdown-item o_pivot_field_selection position-relative"
-                           t-att-data-field="item.fieldName"
-                        >
+                    <Dropdown tag="'li'" togglerClass="'o_menu_item dropdown-item'">
+                        <t t-set-slot="toggler">
                             <t t-esc="item.description"/>
-                        </a>
-                        <div class="dropdown-menu o_inline_dropdown" t-att-class="{ show: openedSubMenus[item.id] }" role="menu">
+                        </t>
                             <t t-foreach="Object.values(intervalOptions)" t-as="option" t-key="option.id">
```
Comment (Polymorphe57) [inline addons/web/static/src/legacy/xml/pivot.xml:100]: identation seems wrong at this level

File: addons/web/static/src/legacy/js/views/pivot/pivot_renderer.js [hunk 10]
```diff
@@ -34,25 +33,25 @@
...
-         */
-        _onToggleCustomGroupbyItem(ev) {
-            if (
-                !ev.target.classList.contains('o_apply_group_by') &&
-                (this.el.contains(ev.target) || this.el.contains(document.activeElement))
-            ) {
-                ev.stopPropagation();
-            }
-        }
+        // /**
```
Comment (Polymorphe57) [inline addons/web/static/src/legacy/js/views/pivot/pivot_renderer.js:36]: commented code here

File: addons/web/static/src/legacy/xml/pivot.xml [hunk 11]
```diff
@@ -60,7 +60,7 @@
                 o_pivot_header_cell_opened: !cell.isLeaf,
             }" t-attf-style="{{ isXAxis ? undefined : 'padding-left: ' + _getPadding(cell) + 'px;' }}" t-att-title="cell.label" t-on-click.self.prevent="_onHeaderClick(cell, isXAxis ? 'col' : 'row')">
             <span t-on-click.self.prevent="_onHeaderClick(cell, isXAxis ? 'col' : 'row')" t-esc="cell.title"/>
-            <t t-if="_isClicked(cell.groupId, isXAxis)" t-call="web.legacy.PivotGroupBySelection"/>
+            <t t-if="cell.isLeaf and !cell.isFolded" t-call="web.legacy.PivotGroupBySelection"/>
```
Comment (Polymorphe57) [inline addons/web/static/src/legacy/xml/pivot.xml:63]: _isClicked should be removed js side

**Comment by Polymorphe57** [comment]
seems good to me except legacy pivot view code that still needs to be cleaned

File: addons/web/static/src/legacy/legacy_setup.js [hunk 12]
```diff
@@ -8,6 +8,7 @@ import {
     makeLegacyDialogMappingService,
     makeLegacyCrashManagerService,
     makeLegacyCommandService,
+    makeLegacyDropdownService,
```
Comment (aab-odoo) [inline addons/web/static/src/legacy/legacy_setup.js:11]: [IMP] web: deploy ui and hotkey as legacy services too

maybe explain why we need this, that it isn't complete (no all new services are there, only the two necessary for the dropdowns)

File: addons/base_import/static/src/legacy/js/import_menu.js [hunk 13]
```diff
@@ -1,18 +1,17 @@
 odoo.define('base_import.ImportMenu', function (require) {
     "use strict";
 
-    const DropdownMenuItem = require('web.DropdownMenuItem');
     const FavoriteMenu = require('web.FavoriteMenu');
     const { useModel } = require('web.Model');
 
+    const { Component } = owl;
```
Comment (aab-odoo) [inline addons/base_import/static/src/legacy/js/import_menu.js:7]: [IMP] web: use new dropdown in legacy too

This commit also needs a bit more explanations of what we're doing, and why

**Review by aab-odoo — COMMENTED** [review]
In second commit message, I'd also say that most of the diff comes from the tests, because such a commit, post freeze, might be scary

**Review by aab-odoo — APPROVED** [review]

**Comment by aab-odoo** [comment]
robodoo rebase-ff r+

**Pull request closed without merging**